packages/core/src/types.ts

```diff
@@ -20,8 +20,11 @@ export interface Lqip {
   attribute?: string;
 }
 
+export type ImageAuto = 'format';
+
 export interface ImageData {
   imageTypes: ImageType[];
+  auto?: ImageAuto;
```

Comment:
I find this type a bit confusing tbh. 

What if we do this?
```
imageTypes: ImageType[] | 'auto';
```

We would not need to search through all array items to look for `'auto'`. But also it would be mutually exclusive with explicit image types, as it should be I think. 

For example in this definition this would be valid type-wise, but still make no sense, right?
```
const imageData = {
  imageTypes: ['png', 'webp'],
  auto: 'format',
  // ...
}
```

Reply:
Made the change in https://github.com/simonihmig/responsive-image/pull/1346/commits/0ddefd4ed01d2e1a19fe1ee4fb8c2cf2df941713

Apart from a small "gotcha" in the Cloudinary function (where [`type` was used to set a file extension](https://github.com/simonihmig/responsive-image/pull/1346/commits/0ddefd4ed01d2e1a19fe1ee4fb8c2cf2df941713#diff-e4a7d7f3654087d08d3d5c5d286130dde01f622aa2e1596d6f9f817df36da854R83)) it works well in the CDN functions.

The new type for `imageTypes` forces some more changes in the component, but nothing huge. And you could argue those changes are wanted anyway.